sio: use `kern.ipc.somaxconn` for `listen()` on Mac (#10141)
tarantool/tarantool: changelogs/unreleased/gh-8130-enable-somaxconn-on-mac.md

## bugfix/core

> * Fixed the bug that on Mac the system setting `kern.ipc.somaxconn` was ignored
>   for listening sockets. Now it is used, but capped at 32367 due to how
>   `listen()` works on Mac (gh-8130).

Reviewer: ```suggestion
* Fixed the bug with the system setting `kern.ipc.somaxconn` ignored on Mac OS
  for listening sockets. Now it is used, but truncated at 32367 due to how
  `listen()` works on Mac (gh-8130).
```

Author: Resolving since the PR is approved.
